### PR TITLE
[EBPF] Generate complexity data in CI

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Enables the generation of complexity data files in KMT CI. It also fixes some errors in the verifier log parser detected during testing of this PR.

This PR is a new attempt to merge this after https://github.com/DataDog/datadog-agent/pull/26723 had to be reverted in https://github.com/DataDog/datadog-agent/pull/27263 due to OOM problems with the VMs. The solution to avoid that problem is reducing the number of platforms where complexity is collected.

As it can be seen in [this dashboard](https://app.datadoghq.com/dashboard/zs9-uia-gsg/ebpf-platform---kernel-matrix-testing---single-pipeline-view?fromUser=false&refresh_mode=paused&tpl_var_ci.pipeline.id%5B0%5D=39025478&tpl_var_git-branch%5B0%5D=%22guillermo.julian%2Fgenerate-complexity-reports%22&view=spans&from_ts=1720769620235&to_ts=1720805196706&live=false), the memory usage with this PR + the [updated test-infra-definitions](https://github.com/DataDog/test-infra-definitions/pull/933) don't generate as much memory pressure as before, with 86% memory usage being the maximum occupation in arm64 instances.

### Motivation

A future PR will leverage these files to report complexity changes in PRs.

### CODE REVIEW

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: .gitlab/kernel_matrix_testing/common.yml [hunk 1]
```diff
@@ -48,6 +48,15 @@
   - $CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $AGENT_QA_PROFILE_SSM_NAME >> ~/.aws/config
   - export AWS_PROFILE=agent-qa-ci
 
+.define_if_collect_complexity:
+  # Collect only from specific platforms to avoid high memory usage
+  - PLATFORMS_FOR_COMPLEXITY_COLLECTION="amazon_5.10 debian_10 ubuntu_23.10 centos_8 opensuse_15.3 fedora_38"
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply .gitlab/kernel_matrix_testing/common.yml]: I replaced ubuntu_23.10 with 18.04 and added SUSE 12.5 to be closer to the minimum supported, but the main problem is that we require 4.15 at least to have the verifier statistics exposed so we cannot get closer.

**Review by usamasaqib — APPROVED** [review]

File: .gitlab/kernel_matrix_testing/common.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (guyarb) [reply .gitlab/kernel_matrix_testing/common.yml]: the minimum of usm is 4.14

**Pull request merged**